tests/app/questionnaire/test_navigator.py

```diff
@@ -114,188 +156,631 @@ def test_routing_conditional_path(self):
+                'group_instance': 0
+            },
+            {
+                "block_id": "96682325-47ab-41e4-a56e-8315a19ffe2a",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 0
+            },
+            {
+                "block_id": "a7dcbb30-1187-4276-a49c-9284730ba4ed",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 0
+            },
+            {
+                "block_id": "96682325-47ab-41e4-a56e-8315a19ffe2a",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 1
+            },
+            {
+                "block_id": "a7dcbb30-1187-4276-a49c-9284730ba4ed",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 1
+            }
+        ]
+
+        answer = Answer(
+            group_id="multiple-questions-group",
+            group_instance=0,
+            answer_id="household-full-name",
+            block_id="household-composition",
+            value="2"
+        )
+        answers = AnswerStore()
+        answers.add(answer)
+
+        navigator = Navigator(survey, answers)
+
+        self.assertEqual(expected_path, navigator.get_routing_path())
+
+    def test_repeating_groups_no_of_answers(self):
+        survey = load_schema_file("test_repeating_household.json")
+
+        expected_path = [
+            {
+                "block_id": "household-composition",
+                "group_id": "multiple-questions-group",
+                'group_instance': 0
+            },
+            {
+                "block_id": "96682325-47ab-41e4-a56e-8315a19ffe2a",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 0
+            },
+            {
+                "block_id": "a7dcbb30-1187-4276-a49c-9284730ba4ed",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 0
+            },
+            {
+                "block_id": "96682325-47ab-41e4-a56e-8315a19ffe2a",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 1
+            },
+            {
+                "block_id": "a7dcbb30-1187-4276-a49c-9284730ba4ed",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 1
+            },
+            {
+                "block_id": "96682325-47ab-41e4-a56e-8315a19ffe2a",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 2
+            },
+            {
+                "block_id": "a7dcbb30-1187-4276-a49c-9284730ba4ed",
+                "group_id": "f22b1ba4-d15f-48b8-a1f3-db62b6f34cc0",
+                'group_instance': 2
+            }
+        ]
+
+        answer = Answer(
+            group_id="multiple-questions-group",
+            group_instance=0,
```

Comment:
Really? Why? Note: The routing rule is updated as part of the test to be based on the value of the answer..